scrapy_poet/page_input_providers.py

```diff
@@ -54,49 +55,39 @@ def __init__(self, response: Response):
         """
 
     @abc.abstractmethod
-    def __call__(self):
+    def __call__(self, provided_classes: typing.Set[typing.Type]) -> typing.Dict[typing.Type, typing.Any]:
         """This method is responsible for building Page Input dependencies."""
 
 
-def register(provider_class: typing.Type[PageObjectInputProvider],
-             provided_class: typing.Type):
-    """This method registers a Page Object Input Provider in the providers
-    registry. It could be replaced by the use of the ``provides`` decorator.
-
-    Example:
-
-        register(ResponseDataProvider, ResponseData)
-    """
-    providers[provided_class] = provider_class
-
-
-def provides(provided_class: typing.Type):
+def provides():
     """This decorator should be used with classes that inherits from
     ``PageObjectInputProvider`` in order to automatically register them as
     providers.
 
     See ``ResponseDataProvider``'s implementation for an example.
     """
     def decorator(provider_class: typing.Type[PageObjectInputProvider]):
-        register(provider_class, provided_class)
+        providers.add(provider_class)
         return provider_class
 
     return decorator
```

Comment:
Returning decorator doesn't make sense now, one can add to providers and return class as-is. I'm not sure decorator interface makes sense with that, but that can be discussed & refactored separately.

Reply:
That was both my initial thought and conclusion: we can deal with it in another pull request.